docs/source/examples/replicate_vicuna/ReplicateVicuna.md

```diff
@@ -0,0 +1,174 @@
+```
+
+
+## Inference
+
+### Concatenation of the checkpoints
+
+As we applied the LoRa method, we first need to merge the finetuned `llama7B-vicuna-onmt.pt` checkpoint in the original `llama7B-onmt.pt` model, using the `lora_weights.py tool`. :
+
+```shell
+python3 OpenNMT-py/tools/lora_weights.py\
+    --action merge \
+    --base_model llama7B-vicuna-onmt \
+    --lora_weights finetuned_llama7B/llama7B-vicuna-onmt_step_4000.pt \
+    --output finetuned_llama7B/llama7B-vicuna-onmt_step_4000.concat.pt
+```
+
+### Input examples
+
+The inputs need to follow the same pattern used in the finetuning examples. 
+
+Let us create an "inputs" folder and save inside it the file named `input_examples.txt`.
+
+
+### Inference with `translate.py`
+
+Let us create an "outputs" folder.
+
+To obtain the model's inference you can run this command:
+
+```shell
+nohup python3 OpenNMT-py/onmt/bin/translate.py\
+    -model finetuned_llama7B/llama7B-vicuna-onmt_step_4000.concat.pt \
+    -src inputs/input_examples.txt \
+    -output outputs/examples_llama7B-vicuna-onmt_step_4000.concat.txt  \
+    -config translate_opts.yaml > infer.log & 
+```
+
+Where `translate_opts.yaml` is the provided config with the translation options.
+You can test other decoding methods and paramaters.
+
+
+### Inference with `CTranslate` 
+
+First we need to do the conversion to the ctranslate2 format.
+
+We have provided a python script to "manually" add the `encoder_type` key to the checkpoint.
+
+```shell
+python3 add_missing_key.py
+
+python3 OpenNMT-py/onmt/bin/release_model.py \
+    --model finetuned_llama7B/llama7B-vicuna-onmt_step_4000.concat_added_key.pt \
+    --output finetuned_llama7B/llama7B-vicuna-onmt_step_4000.concat_CT2 \
+    --format ctranslate2 \
+    --quantization int8_float16
+```
+
```

Comment:
NB: with translate.py the inference output will be detokenized with the extract step which convert back the '\n' to ((newline)) which then can be handled properly depending the use-case.
When using ctranslate2 we need to replace ((newline)) by '\n' in inputs.